src/ifcopenshell-python/ifcopenshell/file.py

```diff
@@ -86,42 +101,75 @@ def __getitem__(self, key):
             return entity_instance(self.wrapped_data.by_guid(str(key)))
 
     def by_id(self, id):
-        """
-        Return IFC objects filtered by IFC ID.
+        """Return IFC objects filtered by IFC ID.
 
-        Returned objects are unwrapped IFC objects.
+        :param id: STEP numerical identifier
+        :type id: int
+        :returns: Returned objects are unwrapped IFC objects.
+        :rtype: ifcopenshell.entity_instance.entity_instance
         """
         return self[id]
 
     def by_guid(self, guid):
-        """
-        Return IFC objects filtered by IFC GUID.
+        """Return IFC objects filtered by IFC GUID.
 
-        Returned objects are unwrapped IFC objects.
+        :param guid: GlobalId value in 22-character encoded form
+        :type guid: string
+        :returns: Returned objects are unwrapped IFC objects.
+        :rtype: ifcopenshell.entity_instance.entity_instance
         """
         return self[guid]
 
     def add(self, inst):
+        """Undocumented function"""
         inst.wrapped_data.this.disown()
         return entity_instance(self.wrapped_data.add(inst.wrapped_data))
 
     def by_type(self, type):
-        """
-        Return IFC objects filtered by IFC Type and wrapped with the entity_instance class.
+        """Return IFC objects filtered by IFC Type and wrapped with the entity_instance class.
 
-        See ifcopenshell.entity_instance for class methods and properties
+        If an IFC type class has subclasses, all entities of those subclasses are also returned.
+
+        :param type: The case insensitive type of IFC class to return.
+        :type type: string
+        :returns: A list of ifcopenshell.entity_instance.entity_instance objects
+        :rtype: list
         """
         return [entity_instance(e) for e in self.wrapped_data.by_type(type)]
 
     def traverse(self, inst, max_levels=None):
+        """Get a list of all referenced entities for a particular entity
```

Comment:
Including the instance `inst`. Generally (and I am a big part in this, we should try to establish consistent wording for instance/class/entity). I would say the entity is the class definition in the schema (although I agree it could mean the instance just as well). Instances (or entity instances) are then the instantiations of these beasts.

Reply:
Fixed. Also made terminology consistent :)